lib/index.js

```diff
@@ -13,5 +13,38 @@ module.exports = (dirname, manifestExtras) => {
 
     const manifest = Manifest.concat(manifestExtras || []);
 
-    return Haute(dirname, manifest);
+    return (server, options, next) => {
+
+        if (typeof options === 'function' ||
+           typeof next !== 'undefined') {
+
+            // callback style
+            if (typeof next !== 'undefined') {
+                return Haute(dirname, manifest)(server, options, next);
+            }
+
+            next = options;
+            return Haute(dirname, manifest)(server, null, next);
+        }
+
+        return new Promise((resolve, reject) => {
+
+            const cb = function (err) {
+
+                if (err) {
+                    reject(err);
```

Comment:
Ideally you'd get haute-couture to try to perform some hapi API call that organically fails.  One example would be using haute-couture to register a plugin that calls `next(err)`.  Or provisioning the same cache twice.  Something like that.